dev-lang/spidermonkey/spidermonkey-68.12.0.ebuild

```diff
@@ -0,0 +1,264 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="7"
+
+PYTHON_COMPAT=( python3_{7..9} )
+
+WANT_AUTOCONF="2.1"
+
+inherit autotools check-reqs python-any-r1 toolchain-funcs
+
+MY_PN="mozjs"
+MY_PV="${PV/_pre*}" # Handle Gentoo pre-releases
+
+MY_MAJOR=$(ver_cut 1)
+
+MOZ_ESR="1"
+
+# Convert the ebuild version to the upstream mozilla version
+MOZ_PV="${MY_PV/_alpha/a}" # Handle alpha for SRC_URI
+MOZ_PV="${MOZ_PV/_beta/b}" # Handle beta for SRC_URI
+MOZ_PV="${MOZ_PV%%_rc*}" # Handle rc for SRC_URI
+
+if [[ ${MOZ_ESR} == 1 ]] ; then
+	# ESR releases have slightly different version numbers
+	MOZ_PV="${MOZ_PV}esr"
+fi
+
+# Patch version
+FIREFOX_PATCHSET="firefox-68.0-patches-15"
+SPIDERMONKEY_PATCHSET="${PN}-68.6.0-patches-04"
+
+MOZ_HTTP_URI="https://archive.mozilla.org/pub/firefox/releases"
+MOZ_SRC_URI="${MOZ_HTTP_URI}/${MOZ_PV}/source/firefox-${MOZ_PV}.source.tar.xz"
+
+if [[ "${PV}" == *_rc* ]]; then
+	MOZ_HTTP_URI="https://archive.mozilla.org/pub/firefox/candidates/${MOZ_PV}-candidates/build${PV##*_rc}"
+	MOZ_SRC_URI="${MOZ_HTTP_URI}/source/firefox-${MOZ_PV}.source.tar.xz"
+fi
+
+PATCH_URIS=(
+	https://dev.gentoo.org/~{anarchy,whissi,polynomial-c,axs}/mozilla/patchsets/${FIREFOX_PATCHSET}.tar.xz
+	https://dev.gentoo.org/~{whissi,polynomial-c,axs}/mozilla/patchsets/${SPIDERMONKEY_PATCHSET}.tar.xz
+)
+
+SRC_URI="${MOZ_SRC_URI}
+	${PATCH_URIS[@]}"
+
+DESCRIPTION="SpiderMonkey is Mozilla's JavaScript engine written in C and C++"
+HOMEPAGE="https://developer.mozilla.org/en-US/docs/Mozilla/Projects/SpiderMonkey"
+
+KEYWORDS="~alpha amd64 arm arm64 ~hppa ~ia64 ~mips ppc ppc64 ~s390 sparc x86"
+
+SLOT="68"
+LICENSE="MPL-2.0"
+IUSE="cpu_flags_arm_neon debug +jit test"
+
+RESTRICT="!test? ( test )"
+
+BDEPEND="dev-lang/python:2.7
+	test? ( ${PYTHON_DEPS} )"
+
+DEPEND=">=dev-libs/icu-63.1:=
```

Comment:
Oh, interesting. 
spidermonkey 68 simply pulls in `dev-libs/icu`, while spidermonkey 60 allows us to set `system-icu` USE flag.
For context, we have recently actually tried to delete `dev-libs/icu` completely, to minimize size of production image.

So it would be really great, if we could delete the dependency on icu here as well.
I see 2 options.
* Set the default version of spidermonkey to 60, and set USE flag like `dev-lang/spidermonkey -system-icu` in `profiles/coreos/base/package.use`.
* Keep spidermonkey 68, and update the ebuild to make it have `system-icu` USE flag just like 60.

Reply:
I have just realized, polkit 0.116 pulls in spidermonkey 60, while polkit 0.117 needs spidermonkey 68.
polkit 0.116 is the default version both for amd and arm.
So the first option, spidermonkey 60 with an additional USE flag would be the way to go.